src/createSharedElementStackNavigator.tsx

```diff
@@ -1,146 +1,193 @@
 import * as React from 'react';
-import hoistNonReactStatics from 'hoist-non-react-statics';
+import {
+  useNavigationBuilder,
+  createNavigatorFactory,
+  StackRouter,
+  DefaultNavigatorOptions,
+  RouteConfig,
+  StackRouterOptions,
+  StackNavigationState,
+} from '@react-navigation/native';
+import {
+  CardAnimationContext,
+  StackView,
+  StackNavigationOptions,
+} from '@react-navigation/stack';
+import { SharedElementRendererProxy } from './SharedElementRendererProxy';
+import SharedElementRendererContext from './SharedElementRendererContext';
 import SharedElementRendererView from './SharedElementRendererView';
-import SharedElementRendererData, {
-  ISharedElementRendererData,
-} from './SharedElementRendererData';
+import SharedElementRendererData from './SharedElementRendererData';
 import createSharedElementScene from './createSharedElementScene';
-import SharedElementRendererContext from './SharedElementRendererContext';
-import { SharedElementRendererProxy } from './SharedElementRendererProxy';
 import {
-  createStackNavigator,
-  CardAnimationContext,
-} from 'react-navigation-stack';
+  SharedElementSceneComponent,
+  SharedElementsComponentConfig,
+} from './types';
+import {
+  StackNavigationConfig,
+  StackNavigationEventMap,
+} from '@react-navigation/stack/lib/typescript/src/types';
 
 let _navigatorId = 1;
 
-function createSharedElementStackSceneNavigator(
-  routeConfigs: any,
-  navigatorConfig: any,
-  rendererData: ISharedElementRendererData,
-  navigatorId: string
-) {
-  //console.log('createSharedElementStackSceneNavigator...', navigatorId);
+export default function createSharedElementStackNavigator<
+  ParamList extends Record<string, object | undefined>
+>(options?: { name?: string }) {
```

Comment:
Question: Is `options` / `name` necessary as an argument here? I reused it from the old implementation, but unsure what its value is.

Reply:
The name is primarily used for debug purposes. Especially with nested navigators it can be very difficult to see where the lifecycle events are coming from. You can turn on some verbose flags and then the `name` is used when logging events for that navigator.